src/web.ts

```diff
@@ -1,22 +1,86 @@
-import { WebPlugin } from '@capacitor/core';
-import { SignInWithApplePlugin } from './definitions';
+import { WebPlugin } from "@capacitor/core";
+
+import {
+  SignInWithApplePlugin,
+  InitOptions,
+  SignInResponse,
+  SignInError,
+} from "./definitions";
+
+declare var window: any;
+
+export class SignInWithAppleWeb extends WebPlugin
+  implements SignInWithApplePlugin {
+  hasInitialized = false;
 
-export class SignInWithAppleWeb extends WebPlugin implements SignInWithApplePlugin {
   constructor() {
     super({
-      name: 'SignInWithApple',
-      platforms: ['web']
+      name: "SignInWithApple",
+      platforms: ["web"],
+    });
+  }
+
+  async Init(options: InitOptions): Promise<void> {
+    this.loadAppleScript(() => {
+      if (window && window.AppleID && !this.hasInitialized) {
+        const { clientId, scope, redirectURI, state, usePopup } = options;
+
+        window.AppleID.auth.init({
+          clientId: clientId,
+          scope: scope,
+          redirectURI: redirectURI,
+          state: state,
+          usePopup: usePopup !== undefined ? usePopup : false,
+        });
+
+        this.hasInitialized = true;
+      }
+    });
+  }
+
+  Authorize(): Promise<SignInResponse> {
+    return new Promise<SignInResponse>(async (resolve, reject) => {
+      try {
+        if (window && window.AppleID) {
+          if (this.hasInitialized) {
+            const response: SignInResponse = await window.AppleID.auth.signIn();
+            resolve(response);
+          } else {
+            reject({
+              error: "AppleID has not yet initialized",
+            } as SignInError);
+          }
+        } else {
+          reject({
+            error: "Cannot find AppleID instance",
+          } as SignInError);
+        }
+      } catch (error) {
+        reject(error as SignInError);
+      }
     });
   }
 
-  async Authorize(): Promise<{response: any}> {
-    return;
+  private loadAppleScript(callback: any) {
+    const file = document.createElement("script");
+    file.setAttribute("type", "text/javascript");
+    file.setAttribute(
+      "src",
+      "https://appleid.cdn-apple.com/appleauth/static/jsapi/appleid/1/en_US/appleid.auth.js"
+    );
```

Comment:
you're still going through the work of creating the element here, even if it's not needed, I'd do something more like:

```
private loadAppleScript(callback: any) {
  if  (window && window.AppleID) {
    return callback()
  }

  const file = ....
}
```

so, return early and save the CPU cycles

Reply:
Refactored in af5ac08